python-checks/src/main/java/org/sonar/python/checks/CheckUtils.java

```diff
@@ -104,4 +119,49 @@ private static boolean calleeHasNameLocals(CallExpression callExpression) {
     Expression callee = callExpression.callee();
     return callee.is(Tree.Kind.NAME) && "locals".equals(((Name) callee).name());
   }
+
+  public static boolean isConstant(Expression condition) {
+    return isImmutableConstant(condition) || isConstantCollectionLiteral(condition);
+  }
+
+  public static boolean isImmutableConstant(Expression condition) {
+    return TreeUtils.isBooleanLiteral(condition) ||
+      condition.is(NUMERIC_LITERAL, STRING_LITERAL, NONE, LAMBDA, GENERATOR_EXPR);
+  }
+
+  public static boolean isConstantCollectionLiteral(Expression condition) {
+    switch (condition.getKind()) {
+      case LIST_LITERAL:
+        return isAlwaysEmptyOrNonEmptyCollection(((ListLiteral) condition).elements().expressions());
+      case DICTIONARY_LITERAL:
+        return isAlwaysEmptyOrNonEmptyCollection(((DictionaryLiteral) condition).elements());
+      case SET_LITERAL:
+        return isAlwaysEmptyOrNonEmptyCollection(((SetLiteral) condition).elements());
+      case TUPLE:
+        return isAlwaysEmptyOrNonEmptyCollection(((Tuple) condition).elements());
+      default:
+        return false;
+    }
+  }
+
+
+
+
+  public static boolean isClassOrFunction(Symbol symbol) {
+    if (symbol.is(Symbol.Kind.CLASS)) {
+      return true;
+    }
+    if (symbol.is(Symbol.Kind.FUNCTION)) {
+      // Avoid potential FPs with properties: only report on limited selection of "safe" decorators
+      return ACCEPTED_DECORATORS.containsAll(((FunctionSymbol) symbol).decorators());
+    }
+    return false;
+  }
+
+  private static boolean isAlwaysEmptyOrNonEmptyCollection(List<? extends Tree> elements) {
```

Comment:
This method name also feels kind of odd - I know you didn't choose it, but it strikes me a bit now that it's in `CheckUtils`.
Probably `doesNotContainUnpackingExpression` is a bit more generic.